packages/block-library/src/post-excerpt/edit.js

```diff
@@ -128,15 +135,18 @@ export default function PostExcerptEdit( {
 	attributes,
 	setAttributes,
 	isSelected,
+	context,
 } ) {
-	if ( ! useEntityId( 'postType', 'post' ) ) {
-		return __( 'Post Excerpt' );
-	}
+	// This check doesn't work in site editor? always false.
+	// if ( ! useEntityId( 'postType', 'post' ) ) {
+	// 	return __( 'Post Excerpt' );
+	// }
```

Comment:
This early return is always getting triggered in the site editor.  Im not sure what else to use as a replacement.

Reply:
Just for reference, for the Post Date (#23931), I've chosen to completely remove the wrapping function, letting the edit function handle the existence of the post attribute.

E.g. (very simplified)

Before
```js
function PostDateEditor( context ) {
  // Get the date with useEntityProp and the post type and ID contained in the context
  return <div>{ date ? date : 'No date' }</div>;
}

export default function PostDateEdit( { context } ) {
  if ( ! context ) { return 'Post date'; }
  return PostDateEditor( context );
}
```

After
```js
export default function PostDateEdit( { context } ) {
  // Get the date with useEntityProp and the post type and ID contained in the context
  return <div>{ date ? date : 'No date' }</div>;
}
```